server/src/main/java/org/opensearch/index/store/InMemoryRemoteSnapshotDirectory.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    @Override
+    public void deleteFile(String name) throws IOException {}
+
+    @Override
+    public IndexOutput createOutput(String name, IOContext context) {
+        return NoopIndexOutput.INSTANCE;
+    }
+
+    @Override
+    public IndexInput openInput(String name, IOContext context) throws IOException {
+        final BlobStoreIndexShardSnapshot.FileInfo fileInfo = fileInfoMap().get(name);
+
+        // Virtual files are contained entirely in the metadata hash field
+        if (fileInfo.name().startsWith("v__")) {
+            return new ByteArrayIndexInput(name, fileInfo.metadata().hash().bytes);
+        }
+
+        try (InputStream is = blobContainer().readBlob(fileInfo.name())) {
+            return new ByteArrayIndexInput(name, is.readAllBytes());
+        }
+    }
+
+    @Override
+    public void close() throws IOException {}
+
+    @Override
+    public long fileLength(String name) throws IOException {
+        initializeIfNecessary();
+        return fileInfoMap.get().get(name).length();
+    }
+
+    @Override
+    public Set<String> getPendingDeletions() throws IOException {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public IndexOutput createTempOutput(String prefix, String suffix, IOContext context) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void sync(Collection<String> names) throws IOException {}
+
+    @Override
+    public void syncMetaData() {}
+
+    @Override
+    public void rename(String source, String dest) throws IOException {}
+
+    @Override
+    public Lock obtainLock(String name) throws IOException {
+        return NoLockFactory.INSTANCE.obtainLock(null, null);
+    }
+
+    static class NoopIndexOutput extends IndexOutput {
+
+        final static NoopIndexOutput INSTANCE = new NoopIndexOutput();
+
+        NoopIndexOutput() {
+            super("noop", "noop");
+        }
+
+        @Override
+        public void close() throws IOException {
+
+        }
+
+        @Override
+        public long getFilePointer() {
+            return 0;
+        }
+
+        @Override
+        public long getChecksum() throws IOException {
+            return 0;
+        }
+
+        @Override
+        public void writeByte(byte b) throws IOException {
+
+        }
+
+        @Override
+        public void writeBytes(byte[] b, int offset, int length) throws IOException {
+
+        }
+    }
+
+    private BlobContainer blobContainer() {
+        initializeIfNecessary();
+        return blobContainer.get();
+    }
+
+    private Map<String, BlobStoreIndexShardSnapshot.FileInfo> fileInfoMap() {
+        initializeIfNecessary();
+        return fileInfoMap.get();
+    }
+
+    /**
+     * Bit of a hack to lazily initialize the blob store to avoid running afoul
+     * of the assertion in {@code BlobStoreRepository#assertSnapshotOrGenericThread}.
+     */
+    private void initializeIfNecessary() {
```

Comment:
Would this work in case of a server restart/recover scenario? 
The thread would not be `SNAPSHOT `in that case right?

Reply:
No, I don't think this will work. This is really just a short-term work around to avoid bringing any changes to that assertion into the scope of this PR.